File: backend/src/main/java/com/cyan/teacher/service/TeacherService.java
package com.cyan.teacher.service;

import com.cyan.teacher.entity.Teacher;

import java.util.List;

public interface TeacherService {

    public Teacher createTeacher(Teacher teacher);
    public List<Teacher> getAllTeachers();
    public Teacher getTeacherById(Long id);
    public Teacher updateTeacher(Long id, Teacher teacher);
    public void deleteTeacher(Long id);


}

File: frontend/src/Services/teacher.service.js
import axios from "axios";

const BASE_API_URL = "http://localhost:8080/api/teacher";

class TeacherService {
  saveTeacher(teacher) {
    return axios.post(BASE_API_URL + "/save", teacher);
  }

  getAllTeacher() {
    return axios.get(BASE_API_URL + "/");
  }

  getTeacherById(id){
    return axios.get(BASE_API_URL+"/"+id);
  }

  deleteTeacher(id){
    return axios.get(BASE_API_URL+"/delete/"+id)
  }

  updateTeacher(id,teacher){
    return axios.post(BASE_API_URL+"/update/"+id,teacher)
  }
}

export default new TeacherService();
